go/libraries/doltcore/sqle/dfunctions/dolt_commit.go

```diff
@@ -41,6 +41,12 @@ func (d DoltCommitFunc) Eval(ctx *sql.Context, row sql.Row) (interface{}, error)
 	dbName := ctx.GetCurrentDatabase()
 	dSess := sqle.DSessFromSess(ctx.Session)
 
+	_, val := dSess.Session.Get(sql.AutoCommitSessionVar)
```

Comment:
I wouldn't check for this specifically. The relevant question isn't whether autocommit is on, but whether there are any changes in the working set (for -a) or staged (without -a). If there aren't, and we haven't set --allow-empty, then a commit will be a no-op and should error out.

Reply:
How could you check for changes in the staged root?